discovery_engine/lib/src/domain/search_manager.dart

```diff
@@ -185,6 +186,21 @@ class SearchManager {
     return EngineEvent.searchTermRequestSucceeded(search.searchTerm);
   }
 
+  /// Return the current trending topics.
+  Future<EngineEvent> trendingTopicsRequested() async {
+    final topics = await _engine.getTrendingTopics();
```

Comment:
:art: it might be good to wrap this in a try-catch and to throw either an `EngineEvent.trendingTopicsRequestFailed()` or an `EngineEvent.engineExceptionRaised()`. but this depends somewhat on the internals of the DE call, which we don't know yet.

Reply:
Fortunately we should be prepared for that already :) If this throws it will be caught above, cause all handlers inside the worker are wrapped and converted to proper `EngineExceptionRaised`.